### DISCUSSION TITLE
编译失败，详情见logs

### DISCUSSION THREAD

**Discussion by lukemin** [Boos4721/openwrt#131] [Bugs/问题反馈]
st will not be visible outside of this definition or declaration
   64 | static inline int batadv_netif_rx(struct sk_buff *skb)
      |                                          ^~~~~~~
/home/ubuntu/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/batman-adv-2022.1/compat-hacks.h: In function 'batadv_netif_rx':
/home/ubuntu/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/batman-adv-2022.1/compat-hacks.h:66:13: error: implicit declaration of function 'in_interrupt' [-Werror=implicit-function-declaration]
   66 |         if (in_interrupt())
      |             ^~~~~~~~~~~~
/home/ubuntu/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/batman-adv-2022.1/compat-hacks.h:67:24: error: implicit declaration of function 'netif_rx' [-Werror=implicit-function-declaration]
   67 |                 return netif_rx(skb);
      |                        ^~~~~~~~
/home/ubuntu/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/batman-adv-2022.1/compat-hacks.h:69:24: error: implicit declaration of function 'netif_rx_ni' [-Werror=implicit-function-declaration]
   69 |                 return netif_rx_ni(skb);
      |                        ^~~~~~~~~~~
In file included from <command-line>:
./include/linux/netdevice.h: At top level:
/home/ubuntu/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/batman-adv-2022.1/compat-hacks.h:71:18: error: conflicting types for 'batadv_netif_rx'; have 'int(struct sk_buff *)'
   71 | #define netif_rx batadv_netif_rx
      |                  ^~~~~~~~~~~~~~~
./include/linux/netdevice.h:4029:5: note: in expansion of macro 'netif_rx'
 4029 | int netif_rx(struct sk_buff *skb);
      |     ^~~~~~~~
/home/ubuntu/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/batman-adv-2022.1/compat-hacks.h:64:19: note: previous definition of 'batadv_netif_rx' with type 'int(struct sk_buff *)'
   64 | static inline int batadv_netif_rx(struct sk_buff *skb)
      |                   ^~~~~~~~~~~~~~~
cc1: some warnings being treated as errors
make[5]: *** [scripts/Makefile.build:277: /home/ubuntu/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/batman-adv-2022.1/net/batman-adv/bat_algo.o] Error 1
make[4]: *** [Makefile:1868: /home/ubuntu/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/batman-adv-2022.1/net/batman-adv] Error 2
make[4]: Leaving directory '/home/ubuntu/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/linux-5.15.34'
make[3]: *** [Makefile:95: /home/ubuntu/openwrt/build_dir/target-aarch64_cortex-a53_musl/linux-ipq807x_generic/batman-adv-2022.1/.built] Error 2
make[3]: Leaving directory '/home/ubuntu/openwrt/feeds/routing/batman-adv'
time: package/feeds/routing/batman-adv/compile#2.33#0.57#3.83
    ERROR: package/feeds/routing/batman-adv failed to build.
make[2]: *** [package/Makefile:116: package/feeds/routing/batman-adv/compile] Error 1
make[2]: Leaving directory '/home/ubuntu/openwrt'
make[1]: *** [package/Makefile:110: /home/ubuntu/openwrt/staging_dir/target-aarch64_cortex-a53_musl/stamp/.package_compile] Error 2
make[1]: Leaving directory '/home/ubuntu/openwrt'
make: *** [/home/ubuntu/openwrt/include/toplevel.mk:230: world] Error 2

**Comment by he110777** [comment]
一样，5月10日更新报错。

**Comment by Boos4721** [comment]
<img width="585" alt="iShot_2022-05-11_14 24 09" src="https://user-images.githubusercontent.com/43845654/167782525-5666e5ee-3ebc-4fe5-a05d-239d4e281165.png">

**Reply by lukemin** [reply]
> <img alt="iShot_2022-05-11_14 24 09" width="585" src="https://user-images.githubusercontent.com/43845654/167782525-5666e5ee-3ebc-4fe5-a05d-239d4e281165.png">
请问你选的设备类型是redmi ax6吗？是不是luci选项 applications选择的原因？

**Reply by supsupsuperstar** [reply]
失败+1 看到上面dl下载报错了，nss-firmware-2022-05-06-20c9a9b7.tar.xz: Download from https://github.com/quic/qca-sdk-nss-fw.git failed ，在线编译下载不了就离谱

**Reply by supsupsuperstar** [reply]
hash还是不对。。。我下了很多次都是56777bd419d86971d0b5d951e64675390411d6ea6ea9baa934577b480010c298

**Comment by lukemin** [comment]
我发现应该是某个luci-applications下面的哪个文件出错了，我用默认设置编译了一下没报错
